servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/tests/BaseTestNessieRest.java

```diff
@@ -399,8 +399,7 @@ public void testInvalidTag(String invalidTagName) {
                         .extract()
                         .as(NessieError.class)
                         .getMessage())
-                .contains(
-                    "Could not resolve type id 'FOOBAR' as a subtype of `org.projectnessie.model.Reference`"));
+                .contains("assignReference.expectedHash: must not be null"));
```

Comment:
Shouldn't this still be something like "Reference type FOOBAR is not supported"? That is to say that we should probably report the same "unsupported" error on the server side regardless of the API method (`assignReference.expectedHash: must not be null` is specific to ref. assignment, but the real problem is that the ref. type is bogus).

Reply:
It's ambiguous with this change to default to `TAG`

Reply:
Ah, right... I guess that default is meaningful only on the client side. If the server received a reference with a `type` that it does not know about, that is probably a "bad request" regardless of how JSON serialization is done... WDYT?

Reply:
That's difficult. Opted to remove the change for `ReferenceType`.